test/metabase/test/data/interface.clj

```diff
@@ -29,19 +29,21 @@
 ;;; |                                   Dataset Definition Record Types & Protocol                                   |
 ;;; +----------------------------------------------------------------------------------------------------------------+
 
-(p.types/defrecord+ FieldDefinition [field-name base-type semantic-type visibility-type fk field-comment])
+(p.types/defrecord+ FieldDefinition [field-name base-type effective-type coercion-strategy semantic-type visibility-type fk field-comment])
 
 (p.types/defrecord+ TableDefinition [table-name field-definitions rows table-comment])
 
 (p.types/defrecord+ DatabaseDefinition [database-name table-definitions])
 
 (def ^:private FieldDefinitionSchema
-  {:field-name                       su/NonBlankString
-   :base-type                        (s/cond-pre {:native su/NonBlankString} su/FieldType)
-   (s/optional-key :semantic-type)   (s/maybe su/FieldType)
-   (s/optional-key :visibility-type) (s/maybe (apply s/enum field/visibility-types))
-   (s/optional-key :fk)              (s/maybe su/KeywordOrString)
-   (s/optional-key :field-comment)   (s/maybe su/NonBlankString)})
+  {:field-name                         su/NonBlankString
+   :base-type                          (s/cond-pre {:native su/NonBlankString} su/FieldType)
+   (s/optional-key :semantic-type)     (s/maybe su/FieldType)
+   (s/optional-key :effective-type)    (s/maybe su/FieldType)
+   (s/optional-key :coercion-strategy) (s/maybe su/CoercionStrategy)
+   (s/optional-key :visibility-type)   (s/maybe (apply s/enum field/visibility-types))
+   (s/optional-key :fk)                (s/maybe su/KeywordOrString)
+   (s/optional-key :field-comment)     (s/maybe su/NonBlankString)})
```

Comment:
Non really part of this PR, but it wouldn't be a bad idea to open this schema up either